src/payment/payjoin.rs

```diff
@@ -0,0 +1,69 @@
+//! Holds a payment handler allowing to send Payjoin payments.
+
+use crate::types::PayjoinSender;
+use crate::{error::Error, Config};
+
+use std::sync::{Arc, RwLock};
+
+/// A payment handler allowing to send Payjoin payments.
+///
+/// Should be retrieved by calling [`Node::payjoin_payment`].
+///
+/// [`Node::payjoin_payment`]: crate::Node::payjoin_payment
+pub struct PayjoinPayment {
+	runtime: Arc<RwLock<Option<tokio::runtime::Runtime>>>,
+	sender: Arc<PayjoinSender>,
+	config: Arc<Config>,
+}
+
+impl PayjoinPayment {
+	pub(crate) fn new(
+		runtime: Arc<RwLock<Option<tokio::runtime::Runtime>>>, sender: Arc<PayjoinSender>,
+		config: Arc<Config>,
+	) -> Self {
+		Self { runtime, sender, config }
+	}
+
+	/// Send an on chain Payjoin transaction as specified in [`BIP77`].
+	///
+	/// ...
+	///
+	///	[`BIP77`]: https://github.com/bitcoin/bips/blob/d7ffad81e605e958dcf7c2ae1f4c797a8631f146/bip-0077.mediawiki
+	pub async fn send(
+		&self, payjoin_uri: payjoin::Uri<'static, bitcoin::address::NetworkUnchecked>,
+		amount: Option<bitcoin::Amount>, fee_rate: Option<bdk::FeeRate>,
+	) -> Result<Option<bitcoin::Txid>, Error> {
+		let rt_lock = self.runtime.read().unwrap();
+		if rt_lock.is_none() {
+			return Err(Error::NotRunning);
+		}
+		let payjoin_uri = match payjoin_uri.require_network(self.config.network) {
+			Ok(uri) => uri,
+			Err(_) => return Err(Error::PayjoinNetworkMismatch),
+		};
+		let sender = Arc::clone(&self.sender);
+		let (mut original_psbt, request, context) =
+			sender.create_payjoin_request(payjoin_uri, amount, fee_rate)?;
+		let response = sender.fetch(&request).await;
```

Comment:
the idea here is that we try to fetch a response first, if we dont get one after the timeout we set on the request, we will start the background process to wait for the receiver response. I think we could emit an event in line 63